src/main/resources/static/assets/js/sdh.js

```diff
@@ -73,8 +75,84 @@
 		    _isShowDetails(field) {
 		    	return field.key === 'show_details';
 		    },
+		    _isEdit(field) {
+		    	return field.key === 'edit';
+		    },
 		    _formatDate(dateString) {
 		    	return dateString.replace(/ \d\d:\d\d:\d\d\.\d/, '');
+		    },
+		    edit(item, index, target) {
+		    	this.modalEditRow.domain = this.domainOptions.filter(it => it.description === item.domain)[0].id;
+		    	if (item.id === "") {
+		    		this.modalEditRow.title = 'New Clinical Review';
+		    	} else {
+		    		this.modalEditRow.title = 'Edit Clinical Review';
+		    		this.modalEditRow.id = item.id;
+		    	      this.modalEditRow.prioritySelected = this.modalEditRow.priorityOptions.filter(it => it.text === item.clinician_priority)[0].value;
+			        this.modalEditRow.readinessSelected = this.modalEditRow.readinessOptions.filter(it => it.text === item.patient_readiness)[0].value;
+		    	}
+				this.$root.$emit('bv::show::modal', 'modalEditRow', target);
+		    },
+		    resetModal() {
+		        this.modalEditRow.title = '';
+		        this.modalEditRow.id = '';
+		        this.modalEditRow.domain = '';
+		        this.modalEditRow.prioritySelected = '';
+		        this.modalEditRow.readinessSelected = '';
+		    },
+		    handleOk(evt) {
+		        // Prevent modal from closing so we can validate
+		        evt.preventDefault();
```

Comment:
Would using [the `.prevent` modifier](https://vuejs.org/v2/guide/events.html#Event-Modifiers) work here?

```
<b-modal id="modalEditRow" ref="modal" @hide="resetModal" :title="modalEditRow.title" @ok.prevent="handleOk">
  ...
</b-modal>
```